lib/ansible/modules/crypto/luks_device.py

```diff
@@ -284,10 +295,41 @@ def run_luks_add_key(self, device, keyfile, new_keyfile):
             raise ValueError('Error while adding new LUKS key to %s: %s'
                              % (device, result[STDERR]))
 
-    def run_luks_remove_key(self, device, keyfile):
+    def run_luks_remove_key(self, device, keyfile, allow_to_remove_last_key=False):
         ''' Remove key from given device
             Raises ValueError when command fails
         '''
+        if not allow_to_remove_last_key:
+            result = self._run_command([self._cryptsetup_bin, 'luksDump', device])
+            if result[RETURN_CODE] != 0:
+                raise ValueError('Error while dumping LUKS header from %s'
+                                 % (device, ))
+            keyslot_count = 0
+            keyslot_area = False
+            keyslot_re = re.compile(r'^Key Slot [0-9]+: ENABLED')
+            for line in result[STDOUT].splitlines():
+                if line.startswith('Keyslots:'):
+                    keyslot_area = True
+                elif line.startswith('  '):
+                    # New versions of "cryptsetup luksDump" use human-readable
```

Comment:
Turns out this is not a new version, but the LUKS2 header dump. (Learned that by looking at the `cryptsetup` sources...)